testsuite/testcases/src/quorum_store_onchain_enable_test.rs

```
@@ -0,0 +1,131 @@
+                /*num_cli_accounts=*/ 0,
+            )
+            .await
+        });
+
+        std::thread::sleep(duration / 2);
+
+        runtime.block_on(async {
+
+            let root_cli_index = cli.add_account_with_address_to_cli(
+                swarm.chain_info().root_account().private_key().clone(),
+                swarm.chain_info().root_account().address(),
+            );
+
+            let current_consensus_config: OnChainConsensusConfig = bcs::from_bytes(
+                &rest_client
+                    .get_account_resource_bcs::<Vec<u8>>(
+                        CORE_CODE_ADDRESS,
+                        "0x1::consensus_config::ConsensusConfig",
+                    )
+                    .await
+                    .unwrap()
+                    .into_inner(),
+            )
+            .unwrap();
+
+            let inner = match current_consensus_config {
+                OnChainConsensusConfig::V1(inner) => inner,
+                OnChainConsensusConfig::V2(_) => panic!("Unexpected V2 config"),
+            };
+
+            // Change to V2
+            let new_consensus_config = OnChainConsensusConfig::V2(ConsensusConfigV1 { ..inner });
+
+            let update_consensus_config_script = format!(
+                r#"
+        script {{
+            use aptos_framework::aptos_governance;
+            use aptos_framework::consensus_config;
+            fun main(core_resources: &signer) {{
+                let framework_signer = aptos_governance::get_signer_testnet_only(core_resources, @0000000000000000000000000000000000000000000000000000000000000001);
+                let config_bytes = {};
+                consensus_config::set(&framework_signer, config_bytes);
+            }}
+        }}
+        "#,
+                generate_blob(&bcs::to_bytes(&new_consensus_config).unwrap())
+            );
+
+            cli.run_script_with_default_framework(root_cli_index, &update_consensus_config_script)
+                .await
```

Would an unwrap() here help fail fast in case the txn doesn't go through?

---

Hmm, it seems the error result bubbles up and fails the test immediately. That seems more graceful than panicking.

---

Makes sense